aws/resource_aws_ec2_transit_gateway_peering_attachment_test.go

```diff
@@ -0,0 +1,405 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func init() {
+	resource.AddTestSweepers("aws_ec2_transit_gateway_peering_attachment", &resource.Sweeper{
+		Name: "aws_ec2_transit_gateway_peering_attachment",
+		F:    testSweepEc2TransitGatewayPeeringAttachments,
+	})
+}
+
+func testSweepEc2TransitGatewayPeeringAttachments(region string) error {
+	client, err := sharedClientForRegion(region)
+	if err != nil {
+		return fmt.Errorf("error getting client: %s", err)
+	}
+	conn := client.(*AWSClient).ec2conn
+	input := &ec2.DescribeTransitGatewayPeeringAttachmentsInput{}
+
+	for {
+		output, err := conn.DescribeTransitGatewayPeeringAttachments(input)
+
+		if testSweepSkipSweepError(err) {
+			log.Printf("[WARN] Skipping EC2 Transit Gateway Peering Attachment sweep for %s: %s", region, err)
+			return nil
+		}
+
+		if err != nil {
+			return fmt.Errorf("error retrieving EC2 Transit Gateway Peering Attachments: %s", err)
+		}
+
+		for _, transitGatewayPeeringAttachment := range output.TransitGatewayPeeringAttachments {
+			if aws.StringValue(transitGatewayPeeringAttachment.State) == ec2.TransitGatewayAttachmentStateDeleted {
+				continue
+			}
+
+			id := aws.StringValue(transitGatewayPeeringAttachment.TransitGatewayAttachmentId)
+
+			input := &ec2.DeleteTransitGatewayPeeringAttachmentInput{
+				TransitGatewayAttachmentId: aws.String(id),
+			}
+
+			log.Printf("[INFO] Deleting EC2 Transit Gateway Peering Attachment: %s", id)
+			_, err := conn.DeleteTransitGatewayPeeringAttachment(input)
+
+			if isAWSErr(err, "InvalidTransitGatewayAttachmentID.NotFound", "") {
+				continue
+			}
+
+			if err != nil {
```

Comment:
Loop should continue after error to sweep as many as possible.
You can use `multierror.Error` from https://github.com/hashicorp/go-multierror.
See for example https://github.com/terraform-providers/terraform-provider-aws/blob/a2373c1b81023e2b3c329905bb0f409c6d63c540/aws/resource_aws_apigatewayv2_api_test.go#L24-L66

Reply:
https://github.com/terraform-providers/terraform-provider-aws/pull/11162#issuecomment-611394615